libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1489,6 +1489,23 @@ bool TypeChecker::visit(FunctionCall const& _functionCall)
 		}
 	}
 
+	if (auto memberAccess = dynamic_cast<MemberAccess const*>(&_functionCall.expression()))
+		if (auto contractType = dynamic_cast<ContractType const*>(type(memberAccess->expression()).get()))
+			if (auto callType = dynamic_cast<FunctionType const*>(type(*memberAccess).get()))
+			{
+				auto kind = callType->kind();
+				auto fallbackFunction = contractType->contractDefinition().fallbackFunction();
+
+				if (
+					(kind == FunctionType::Kind::Send || kind == FunctionType::Kind::Transfer) &&
+					!(fallbackFunction && fallbackFunction->isPayable())
```

Comment:
It is a personal taste, but for me `(!fallbackFunction || !fallbackFunction->isPayable())` seems more readable.